### PR TITLE
Experiment with Prettier

### PR DESCRIPTION
**Pull request by kum-deepak**
I have been trying to use `tslint` to help maintain consistent code formatting. However, after going through all options, it still leaves formatting inconsistencies.

We have started using Prettier (https://prettier.io/) in some of our projects. On new projects, it seems quite natural. I wanted to experiment and see how does it look on this codebase.

The default settings cause too many changes. As a philosophy, Prettier supports very few options. I found the following goes inline with current code conventions:

```JSON
{
  "arrowParens": "avoid",
  "printWidth": 100,
  "singleQuote": true,
  "tabWidth": 4
}
```

It does bring consistent formatting. One departure from current code conventions (no space before `()` in function declarations):

```typescript
 public calculateRadiusDomain (): void {
```
to
```typescript
 public calculateRadiusDomain(): void {
```

I have put two commits here, first one is the setup and the second has some sample conversions. I will also add some comments against specific code lines to highlight some issues.

If we decide to go this route we can pass our tests and samples as well.

### CODE REVIEW

File: src/charts/bar-chart.ts [hunk 1]
```diff
@@ -177,19 +184,23 @@ export class BarChart extends StackMixin {
...
             .attr('class', 'bar')
             .attr('fill', (d, i) => this.getColor(d, i))
             .attr('x', d => this._barXPos(d))
             .attr('y', this.yAxisHeight())
             .attr('height', 0);
 
-        const barsEnterUpdate: Selection<SVGRectElement, unknown, SVGGElement, any> = enter.merge(bars);
+        const barsEnterUpdate: Selection<SVGRectElement, unknown, SVGGElement, any> = enter.merge(
+            bars
+        );
```
Comment (kum-deepak) [inline src/charts/bar-chart.ts:204]: This formatting looks weird. These can be forced with `// prettier-ignore`

Comment (gordonwoodhull) [reply src/charts/bar-chart.ts:204]: Is this because the line was too long?

**Comment by gordonwoodhull** [comment]
I'm okay with changing to prettier. Weren't we getting more out of tslint than just formatting, though? Like... actual linting? I'm confused.

File: src/charts/bar-chart.ts (hunk 1 above)
Comment (kum-deepak) [reply src/charts/bar-chart.ts:204]: Prettier is different than linters. It tries to optimize for human readers, see thier note https://prettier.io/docs/en/options.html#print-width

**Comment by kum-deepak** [comment]
`ts-lint` will still be used. Prettier does not support linting actually. I am also working on tightening `ts-lint` as much as I can. In that effort, I was hitting limitations on formatting and I thought to try prettier.

I will go ahead and configure these to work together. Once ready I will force push into this branch.

File: src/charts/bar-chart.ts (hunk 1 above)
Comment (gordonwoodhull) [reply src/charts/bar-chart.ts:204]: I can't imagine they are doing this for readability. The original line is 106 characters.

Comment (kum-deepak) [reply src/charts/bar-chart.ts:204]: :smile: I think they would get better over time.

**Comment by kum-deepak** [comment]
`grunt lint` will run prettier in check mode, and `grunt lint-fix` in write mode. Currently only covering `src`.

**Pull request merged**